app/forms/supplementary/ProcedureCodes.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.supplementary
+
+import play.api.data.Forms.{seq, text}
+import play.api.data.{Form, Forms}
+import play.api.libs.json.Json
+import utils.validators.FormFieldValidator._
+
+case class ProcedureCodes(
+  procedureCode: String,      // max 4 alphanumeric characters
+  additionalProcedureCodes: Seq[String]     // max 99 codes, each is max 3 alphanumeric characters
+) {
+
+  def toMetadataProperties(): Map[String, String] = {
```

Comment:
Probably we should move all methods `toMetadataProperties` to class, only address needs to have it in a companion object, we can add it to our TODO list